### PR TITLE
Block using rules with hypertables

### PR DESCRIPTION
**Pull request by cevian**

### CODE REVIEW

File: src/hypertable.c [hunk 1]
```diff
@@ -812,6 +812,16 @@ table_has_tuples(Oid table_relid, Snapshot snapshot, LOCKMODE lockmode)
 	return hastuples;
 }
 
+static bool
+table_has_rules(Oid table_relid, LOCKMODE lockmode)
+{
+	Relation	rel = heap_open(table_relid, lockmode);
```
[the remark this answers is not in the document]
Comment (cevian) [reply src/hypertable.c]: Yeah I had the same thought but wanted to do it all at once in a refactor after all of these PRs are in.

File: src/hypertable.c [hunk 2]
```diff
@@ -960,6 +970,12 @@ hypertable_create(PG_FUNCTION_ARGS)
 				 errmsg("table \"%s\" is already partitioned", get_rel_name(table_relid)),
 				 errdetail("It is not possible to turn tables that use inheritance into hypertables.")));
 
+	if (table_has_rules(table_relid, AccessShareLock))
+			ereport(ERROR,
+				(errcode(ERRCODE_FEATURE_NOT_SUPPORTED),
+				 errmsg("table \"%s\" has attached rules", get_rel_name(table_relid)),
```
[the remark this answers is not in the document]
Comment (cevian) [reply src/hypertable.c]: Fixed

File: src/process_utility.c [hunk 3]
```diff
@@ -1730,12 +1753,15 @@ process_ddl_command_start(ProcessUtilityArgs *args)
...
+		case T_RuleStmt:
+			process_create_rule_start(args->parsetree);
+			break;
 		case T_DropStmt:
 
 			/*
 			 * Drop associated metadata/chunks but also continue on to drop
 			 * the main table. Because chunks are deleted before the main
-			 * table is dropped, the drop respects CASCADE in the expected
+			 * tabe is dropped, the drop respects CASCADE in the expected
```
[the remark this answers is not in the document]
Comment (cevian) [reply src/process_utility.c]: Fixed

**Comment by cevian** [comment]
@erimatnor  ready for review

File: src/hypertable.c [hunk 4]
```diff
@@ -836,6 +836,12 @@ table_has_replica_identity(Relation rel)
 	return rel->rd_rel->relreplident != REPLICA_IDENTITY_DEFAULT;
 }
 
+static bool
+table_has_rules(Relation rel)
+{
+	return rel->rd_rules != NULL;
```
Comment (erimatnor) [inline src/hypertable.c]: Maybe a simple NULL check like this doesn't warrant a full function, but instead fits better as a macro, or at least an inline function?

**Pull request merged**